### ISSUE TITLE
Inconsistent nuget package for release 3.1.1

### ISSUE THREAD

**Issue by cello99d** [dotnet/efcore#19959]
After upgrading from 3.1.0 to 3.1.1 I get the following exception on app startup

```
System.IO.FileLoadException
  HResult=0x80131040
  Message=Die Datei oder Assembly "Microsoft.Extensions.DependencyInjection.Abstractions, Version=3.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60" oder eine Abhängigkeit davon wurde nicht gefunden. Die gefundene Manifestdefinition der Assembly stimmt nicht mit dem Assemblyverweis überein. (Ausnahme von HRESULT: 0x80131040)
  Source=Microsoft.EntityFrameworkCore
  StackTrace:
   at Microsoft.EntityFrameworkCore.Internal.ServiceProviderCache.<>c__DisplayClass4_0.<GetOrAdd>g__BuildServiceProvider|3()
   at Microsoft.EntityFrameworkCore.Internal.ServiceProviderCache.<>c__DisplayClass4_0.<GetOrAdd>b__2(Int64 k)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.EntityFrameworkCore.Internal.ServiceProviderCache.GetOrAdd(IDbContextOptions options, Boolean providerRequired)
   at Microsoft.EntityFrameworkCore.DbContext..ctor(DbContextOptions options)
   at Microsoft.EntityFrameworkCore.DbContext..ctor()
```

 I need to redirect the following bindings from 3.1.0. to 3.1.1 to make 3.1.1 working:
```xml
<dependentAssembly>
	<assemblyIdentity name="Microsoft.Extensions.DependencyInjection.Abstractions" publicKeyToken="adb9793829ddae60" culture="neutral" />
	<bindingRedirect oldVersion="3.1.0.0" newVersion="3.1.1.0" />
</dependentAssembly>
<dependentAssembly>
	<assemblyIdentity name="Microsoft.Extensions.Logging.Abstractions" publicKeyToken="adb9793829ddae60" culture="neutral" />
	<bindingRedirect oldVersion="3.1.0.0" newVersion="3.1.1.0" />
</dependentAssembly>	
<dependentAssembly>
	<assemblyIdentity name="Microsoft.Extensions.Options" publicKeyToken="adb9793829ddae60" culture="neutral" />
	<bindingRedirect oldVersion="3.1.0.0" newVersion="3.1.1.0" />
</dependentAssembly>		
<dependentAssembly>
	<assemblyIdentity name="Microsoft.Extensions.Caching.Abstractions" publicKeyToken="adb9793829ddae60" culture="neutral" />
	<bindingRedirect oldVersion="3.1.0.0" newVersion="3.1.1.0" />
</dependentAssembly>			
<dependentAssembly>
	<assemblyIdentity name="Microsoft.Extensions.Primitives" publicKeyToken="adb9793829ddae60" culture="neutral" />
	<bindingRedirect oldVersion="3.1.0.0" newVersion="3.1.1.0" />
</dependentAssembly>						`
```

**Comment by AndriySvyryd** [comment]
@Pilchie

**Comment by Pilchie** [comment]
Thanks - the issue here is that the reference assemblies for those are fixed at assembly version 3.1.0.0 for the .NET Core use case, but that prevents automatic binding redirect generation.  We have heard a proposal to multi-target to .NET Framework so that the .NET Framework assets have the new, unique version, but we haven't implemented that.  We might consider it as part of .NET 5, but we're not committed yet.
